Expensify/App
# Refactor IOU Components to /ReportActionItem folder & rename

CC @tgolen -- [as you suggested this change](https://github.com/Expensify/Expensify.cash/pull/2394#discussion_r631918575) in an older PR review

### Details
Simply refactors the three `ReportActionItemIOU...` components to a new package to simplify the naming. The components are `IOUPreview`, `IOUAction`, and `IOUQuote` and all relate to the Preview component -- which is what we should test.
### Fixed Issues
Fixes https://github.com/Expensify/Expensify/issues/164356

### Tests

We need to review the IOU flows to ensure the regression didn't break anything. Please do the following and raise any issues that occur:
- Create a new IOU Request of any amount
- Create a new Bill Split of any amount with 1 user
- Ensure you see a preview component in the chat

![Screenshot 2021-06-18 at 14 18 34](https://user-images.githubusercontent.com/10736861/122566997-17295080-d040-11eb-96d2-08d118a1f0f3.png)

- View Details for one of these requests and ensure you see the Preview component
- Cancel/decline one of the transactions transaction

![Screenshot 2021-06-18 at 14 16 03](https://user-images.githubusercontent.com/10736861/122566680-bc8ff480-d03f-11eb-8819-fba315b28f50.png)

- As the receiving user, settle one of the IOU's using the 'I'll settle elsewhere' option
- The Preview Component should show 'user **PAID** user' in both the View Details and chat views

![Screenshot 2021-06-18 at 14 22 47](https://user-images.githubusercontent.com/10736861/122567618-c1a17380-d040-11eb-9318-ea2b41088087.png)


### QA Steps


### Tested On


### Screenshots

![Screenshot 2021-06-18 at 14 16 03](https://user-images.githubusercontent.com/10736861/122567714-df6ed880-d040-11eb-8c81-665e6691421a.png)


#### Web

![Screenshot 2021-06-18 at 14 18 34](https://user-images.githubusercontent.com/10736861/122570670-e1866680-d043-11eb-90c5-5e51e790a793.png)


#### Mobile Web

![Simulator Screen Shot - iPhone 8 - 2021-06-18 at 14 45 23](https://user-images.githubusercontent.com/10736861/122570647-d7fcfe80-d043-11eb-9459-7662dc5704cc.png)


#### Desktop

<img width="1198" alt="Screenshot 2021-06-18 at 14 34 53" src="https://user-images.githubusercontent.com/10736861/122570709-e9460b00-d043-11eb-8b77-e1350a3fb965.png">


#### iOS

![Simulator Screen Shot - iPhone 8 - 2021-06-18 at 14 44 26](https://user-images.githubusercontent.com/10736861/122570584-c9164c00-d043-11eb-8195-66ab3a70f660.png)


#### Android

![Screenshot_1623334567](https://user-images.githubusercontent.com/10736861/122571266-74bf9c00-d044-11eb-8120-74e35d8ffc9c.png)
